### ISSUE TITLE
Log roll over causes active users stat to become inaccurate

### ISSUE THREAD

**Issue by willnx** [willnx/vlab#22]
On each user's gateway, when the log file that holds the messages for forward packets is rolled over, the [tail](https://github.com/willnx/vlab_ipam_api/blob/master/vlab_ipam_api/log_sender.py#L18) logic results in no new messages being forwarded. The end result is that the Active Users stat is no longer accurate.

**Comment by willnx** [comment]
I wonder if I can just add a line here to HUP the log_sender.py process 🤔 
I would rather have that process be notified _externally_ that the file was rotated, then handle processing that signal.

```
administrator@willhn:/etc/logrotate.d$ cat /usr/lib/rsyslog/rsyslog-rotate
#!/bin/sh

if [ -d /run/systemd/system ]; then
    systemctl kill -s HUP rsyslog.service
else
    invoke-rc.d rsyslog rotate > /dev/null
fi
```

Looks like the rotation is just a cronjob

```
administrator@willhn:/etc/logrotate.d$ cat /etc/cron.daily/logrotate
#!/bin/sh

# Clean non existent log file entries from status file
cd /var/lib/logrotate
test -e status || touch status
head -1 status > status.clean
sed 's/"//g' status | while read logfile date
do
    [ -e "$logfile" ] && echo "\"$logfile\" $date"
done >> status.clean
mv status.clean status

test -x /usr/sbin/logrotate || exit 0
/usr/sbin/logrotate /etc/logrotate.conf
```

And the config to rotate logs includes rsyslog
```
administrator@willhn:/etc/logrotate.d$ cat rsyslog
/var/log/syslog
{
        rotate 7
        daily
        missingok
        notifempty
        delaycompress
        compress
        postrotate
                /usr/lib/rsyslog/rsyslog-rotate
        endscript
}

/var/log/mail.info
/var/log/mail.warn
/var/log/mail.err
/var/log/mail.log
/var/log/daemon.log
/var/log/kern.log
/var/log/auth.log
/var/log/user.log
/var/log/lpr.log
/var/log/cron.log
/var/log/debug
/var/log/messages
{
        rotate 4
        weekly
        missingok
        notifempty
        compress
        delaycompress
        sharedscripts
        postrotate
                /usr/lib/rsyslog/rsyslog-rotate
        endscript
}
```

and the `/var/log/kern.log` file is managed by rsyslog:
```
administrator@willhn:/etc/rsyslog.d$ grep kern 50-default.conf
kern.*                          -/var/log/kern.log
```

**Comment by willnx** [comment]
Handling the rotation in the `log_sender` module itself was waaay simpler.
That code has been pushed to PyPi, and now all I have to do is:

1) update the existing user's gateways
2) update the OVA template (for when new gateways are deployed)

**Comment by willnx** [comment]
Oh geeze, that was more work than I originally anticipated...
Updating the OVA was pretty easy.

Updating existing user's gateways; that was _"fun"_. I needed to add `libpq-dev` to all the user's gateways before I could update `vlab-ipam-api`, so I figured that I should probably run a pile of system updates too. Turns out, SaltStack provide a fully interactive shell, which is great until a package update requires a user to respond with yes or no. That's all settled now, and every user's gateway has the updated log-sender process running. 

I'm going to keep this issue open for a little bit longer, just to ensure that I start seeing active users after a few days.
